### PR TITLE
rbd-mirror: persist image map timestamp

### PR DESCRIPTION
**Pull request by vshankar**

### CODE REVIEW

**Comment by vshankar** [comment]
@ceph-jenkins retest this pease

**Comment by vshankar** [comment]
> @vshankar Can't break the ABI of the structures

yeh -- but since this is not integrated with `rbd-mirror` yet,  there are no consumers of this as of now. Wouldn't that be ok?

**Comment by dillaman** [comment]
@vshankar I guess I thought it was already in the luminous branch. However, you do need to fix the unittest failure due to the encode/decode of the type.

File: src/cls/rbd/cls_rbd_types.cc [hunk 1]
```diff
@@ -506,22 +510,23 @@ void MirrorImageMap::generate_test_instances(
...
-  o.push_back(new MirrorImageMap("uuid-123", data));
-  o.push_back(new MirrorImageMap("uuid-abc", data));
+  o.push_back(new MirrorImageMap("uuid-123", ceph_clock_now(), data));
+  o.push_back(new MirrorImageMap("uuid-abc", ceph_clock_now(), data));
 }
 
 bool MirrorImageMap::operator==(const MirrorImageMap &rhs) const {
-  return instance_id == rhs.instance_id && data.contents_equal(data);
+  return instance_id == rhs.instance_id && mapped_time == rhs.mapped_time &&
+    data.contents_equal(data);
```
Comment (dillaman) [inline src/cls/rbd/cls_rbd_types.cc]: Nit: ```data.contents_equal(rhs.data)```

File: src/cls/rbd/cls_rbd_types.cc [hunk 2]
```diff
@@ -506,22 +510,23 @@ void MirrorImageMap::generate_test_instances(
...
 }
 
 bool MirrorImageMap::operator<(const MirrorImageMap &rhs) const {
-  return  instance_id < rhs.instance_id;
+  return instance_id < rhs.instance_id || mapped_time < rhs.mapped_time;
```
Comment (dillaman) [inline src/cls/rbd/cls_rbd_types.cc]: Needs to be consistently ordered, so it should compare ```instance_id``` first, then if equal compare ```mapped_timed```

**Comment by vshankar** [comment]
build failures are unrelated to this change.

**Comment by dillaman** [comment]
retest this please

**Pull request merged**